generate.py

```diff
@@ -1,23 +1,36 @@
 import pandas as pd
+import requests
 from jinja2 import Environment, FileSystemLoader
 
+good_reads_api_key = input("Enter Goodreads Api Key: ")
 
 books_df = pd.read_excel("./input/Free+English+textbooks.xlsx")
 grouped_books_df = books_df.groupby(["English Package Name"])
 
-
 loader = FileSystemLoader(searchpath="./templates/")
 env = Environment(loader=loader)
 template = env.get_template("index.html")
 
+isbn_hyphen_maps = [{}, {}]
+isbn_review_map = {}
+
+for book in books_df.iterrows():
+    isbn_hyphen_maps[0][book[1]["Print ISBN"]] = book[1]["Print ISBN"].replace('-', '')
+    isbn_hyphen_maps[1][book[1]["Electronic ISBN"]] = book[1]["Electronic ISBN"].replace('-', '')
+for map in isbn_hyphen_maps:
+    response = requests.get("https://www.goodreads.com/book/review_counts.json?isbns=" + ','.join(map.values()) + 'key=' + good_reads_api_key)
+    for review in response.json()["books"]:
+        isbn_review_map[review["isbn13"]] = review
+
 grouped_books = {}
 
 for group, books in grouped_books_df:
     grouped_books[group] = []
     for book in books.iterrows():
+        book[1]["goodreads_review"] = isbn_review_map.get(isbn_hyphen_maps[0][book[1]["Print ISBN"]], isbn_review_map.get(isbn_hyphen_maps[1][book[1]["Electronic ISBN"]], None))
         grouped_books[group].append(book)
 
 rendered_template = template.render(grouped_books=grouped_books)
 
-with open("index.html", "w") as f:
-    f.write(rendered_template)
+with open("index.html", "w", encoding="utf-8") as f:
```

Comment:
Is this not the default? And what was failing without this encoding being set?

Reply:
Received this error on Windows 10, Pythong 3.8.1, Visual Studio 2019
![image](https://user-images.githubusercontent.com/7141342/85122446-6c3b4800-b1db-11ea-8b1b-580443f406ac.png)